crates/ruff_python_formatter/src/statement/stmt_for.rs

```diff
@@ -1,12 +1,75 @@
-use crate::{not_yet_implemented, FormatNodeRule, PyFormatter};
+use crate::comments::{leading_alternate_branch_comments, trailing_comments};
+use crate::expression::parentheses::Parenthesize;
+use crate::prelude::*;
+use crate::{FormatNodeRule, PyFormatter};
 use ruff_formatter::{write, Buffer, FormatResult};
-use rustpython_parser::ast::StmtFor;
+use ruff_python_ast::node::AstNode;
+use rustpython_parser::ast::{Ranged, Stmt, StmtFor};
 
 #[derive(Default)]
 pub struct FormatStmtFor;
 
 impl FormatNodeRule<StmtFor> for FormatStmtFor {
     fn fmt_fields(&self, item: &StmtFor, f: &mut PyFormatter) -> FormatResult<()> {
-        write!(f, [not_yet_implemented(item)])
+        let StmtFor {
+            range: _,
+            target,
+            iter,
+            body,
+            orelse,
+            type_comment: _,
+        } = item;
+
+        let comments = f.context().comments().clone();
+        let dangling_comments = comments.dangling_comments(item.as_any_node_ref());
+        let body_start = body.first().map_or(iter.end(), Stmt::start);
+        let or_else_comments_start =
+            dangling_comments.partition_point(|comment| comment.slice().end() < body_start);
+
+        let (trailing_condition_comments, or_else_comments) =
+            dangling_comments.split_at(or_else_comments_start);
+
+        write!(
+            f,
+            [
+                text("for"),
+                space(),
+                // TODO: the `IfBreaks` is currently ignored by
+                // https://github.com/astral-sh/ruff/blob/4b9b6829dccabdd4faf6efa6a118b4868347a701/crates/ruff_python_formatter/src/expression/expr_tuple.rs#L78
```

Comment:
We might need a new setting that actually removes expression parentheses if not used, i'll experiment with this

Reply:
do we want to keep this open pending that or merge as is for now?